aspnetcore/migration/22-to-30.md

```diff
@@ -69,3 +69,22 @@ Newtonsoft settings can be set with `AddNewtonsoftJson`:
                options.SerializerSettings.ContractResolver = 
                   new CamelCasePropertyNamesContractResolver());
     ```
+
+## Generic host replaces WebHostBuilder
+
+The ASP.NET Core 3.0 templates use [Generic Host](fundamentals/host/generic-host). Previous versions used [WebHostBuilder](xref:fundamentals/host/web-host). The following code shows the ASP.NET Core 3.0 template generated `Program` class:
+
+[!code-csharp[Main](22-to-30/samples/Program.cs?name=snippet)]
+
+The following code shows the ASP.NET Core 2.2 template generated `Program` class:
+
+[!code-csharp[Main](22-to-30/samples/Program2.2.cs?name=snippet)]
+
+<xref:Microsoft.AspNetCore.Hosting.IWebHostBuilder> is used with the <xref:Microsoft.AspNetCore.Hosting.WebHostBuilder> (in the 2.2 templates) and with `webBuilder` (in the 3.0 templates). `WebHostBuilder` will be deprecated in a future release, and eventually removed.  `HostBuilder` will replace `WebHostBuilder`.
+
+## Moving from WebHostBuilder to HostBuilder
+
+The most significant change from `WebHostBuilder` to `HostBuilder` is in [Dependency injection](xref:fundamentals/dependency-injection) (DI). `HostBuilder` can only inject <xref:Microsoft.Extensions.Configuration.IConfiguration> and <xref:Microsoft.Extensions.Hosting.IHostingEnvironment>. The DI limitations:
+
+* Enable the DI container to built once.
+* Avoid several other DI issues.
```

Comment:
Nonspecific ... I guess engineering didn't provide details. Wish for examples or something.